partner_street_number/__init__.py

```diff
@@ -33,11 +33,11 @@ def post_init_hook(cr, registry):
     """
     logging.getLogger('openerp.addons.partner_street_number').info(
         'Migrating existing street names')
-    cr.execute(
-        'SELECT id, street FROM res_partner '
-        'WHERE street IS NOT NULL and street_name IS NULL'
-        )
-    partner_obj = registry['res.partner']
-    for partner in cr.fetchall():
-        partner_obj.write(
-            cr, SUPERUSER_ID, partner[0], {'street': partner[1]})
+
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    partners = env['res.partner'].search(
```

Comment:
add `.with_context(active_test=False)` to get all partners here.
Search can be formatted more beautifully as

```python
myobject.search([
    (prop1),
    (prop2),
])
```

making a change in a proposition a one line change